--- /home/dpetrov/develop/python/pyneurgen/pyneurgen/genotypes.py	2017-04-04 13:17:23.329717680 +0300
+++ genotypes.py	2017-04-04 17:30:04.293123932 +0300
@@ -41,7 +41,7 @@
 TIMEOUT_PROG_EXECUTE = 1
 
 DEFAULT_LOG_FILE = 'pyneurgen.log'
-DEFAULT_LOG_LEVEL = logging.INFO
+DEFAULT_LOG_LEVEL = logging.DEBUG
 
 logging.basicConfig(format='%(asctime)s %(message)s',
                     filename=DEFAULT_LOG_FILE,
